src/native-common/WebView.tsx

```diff
@@ -109,9 +119,50 @@ export class WebView extends React.Component<RX.Types.WebViewProps, RX.Types.Sta
         return undefined;
     }
 
+    private _buildInjectedJavascript(): string {
+        // Keep compatibility with old code that uses window.postMessage. For more information,
+        // see https://github.com/react-native-community/react-native-webview/releases/tag/v5.0.0
+        let injectedJavascript = `
+            // WebView -> Native
+            (function() {
+                if (window.ReactNativeWebView && window.ReactNativeWebView.postMessage) {
+                    window.postMessage = function(data) {
+                        window.ReactNativeWebView.postMessage(data);
+                    };
+                }
+            })();
+
+            // Native -> WebView
+            (function() {
+                window.postMessageFromReactXP = function(message) {
+                    var event;
+                    try {
+                        event = new MessageEvent('message', { data: message });
+                    } catch (e) {
+                        event = document.createEvent('MessageEvent');
+                        event.initMessageEvent('message', true, true, data.data, data.origin, data.lastEventId, data.source);
+                    }
+                    document.dispatchEvent(event);
+                };
+            })();`;
+
+        if (this.props.injectedJavaScript !== undefined) {
+            injectedJavascript += this.props.injectedJavaScript;
+        }
+
+        // End the injectedJavascript with 'true;' or else you'll sometimes get silent failures
+        if (injectedJavascript[injectedJavascript.length - 1] === ';') {
```

Comment:
FYI you don't have to do this check, you can always append ```;true;``` since ```;;``` is a noop in JS